llvm/lib/Passes/StandardInstrumentations.cpp

```diff
@@ -1495,7 +1496,7 @@ void TimeProfilingPassesHandler::registerCallbacks(
 }
 
 void TimeProfilingPassesHandler::runBeforePass(StringRef PassID, Any IR) {
-  timeTraceProfilerBegin(PassID, getIRName(IR));
+  timeTraceProfilerBegin(PassID, demangle(getIRName(IR)));
```

Comment:
This is a bit brute-force as it also runs demangle on cases like `[module]`.
Also for SCCs it fails to demangle since getName returns something like `(func1[,func2])`.
In general those function entries are duplicated a lot in the trace file, maybe one could limit it to certain levels in the flame graph, e.g. only once for `PassManager<Function>` and not for every single pass beneath it like `EarlyCSEPass`.

Reply:
getIRName is a locally defined function (see above in this file).  In most of the cases, it is desirable to have the mangled name because that can be used for filtering.  In this case, I don't think that there is filtering available, so demangling the name seems reasonable.  However, as you noted, there are many situations where demangling is not appropriate.  I would suggest that you add a parameter to the getIRName function (defaulting to false) to allow demangling of appropriate names and do the demangling there.